### PR TITLE
Fix e2e test while combining data from multiple maintenaces 

### PR DESCRIPTION
**Pull request by chadell**

### CODE REVIEW

File: tests/unit/test_e2e.py [hunk 1]
```diff
@@ -849,10 +859,18 @@ def test_provider_get_maintenances(
...
+            # TODO: Tests assume that maintenances (multiple) will be discovered on the first parser
+            if not expected_result and isinstance(partial_result_data, list):
                 expected_result = partial_result_data
+
+            if expected_result and isinstance(partial_result_data, dict):
+                for _ in range(len(expected_result)):
+                    expected_result[0].update(partial_result_data)
             else:
-                expected_result[0].update(partial_result_data[0])
+                assert len(expected_result) == len(expected_result)
```
Comment (scetron) [inline tests/unit/test_e2e.py:871]: Would this not always be true?

Comment (chadell) [reply tests/unit/test_e2e.py:871]: if the expected results are well defined, yes :)

Comment (chadell) [reply tests/unit/test_e2e.py]: you're right, typo there

**Comment by scetron** [comment]
This looks good to me!

**Comment by chadell** [comment]
> This looks good to me!

if so, could you approve the PR?

**Review by scetron — APPROVED** [review]

**Pull request merged**